src/testsuite/vec_f32_dummy.c

```diff
@@ -33,6 +33,304 @@
+}
+
+vf64_t
+test_vec_vglfsso (float *array, const long long offset0,
+		     const long long offset1)
+{
+  return vec_vglfsso (array, offset0, offset1);
+}
+
+vf64_t
+test_vec_vglfsso_032 (float *array)
+{
+  return vec_vglfsso (array, 0, 32);
+}
+
+vf64_t
+test_vec_vlxsspx (const signed long long ra, const float *rb)
+{
+  return vec_vlxsspx (ra, rb);
+}
+
+vf64_t
+test_vec_vlxsspx_c0 (const float *rb)
+{
+  return vec_vlxsspx (0, rb);
+}
+
+vf64_t
+test_vec_vlxsspx_c1 (const float *rb)
+{
+  return vec_vlxsspx (8, rb);
+}
+
+vf64_t
+test_vec_vlxsspx_c2 (const float *rb)
+{
+  return vec_vlxsspx (32760, rb);
+}
+
+vf64_t
+test_vec_vlxsspx_c3 (const float *rb)
+{
+  return vec_vlxsspx (32768, rb);
+}
+
+vf64_t
+test_vec_vlxsspx_c4 (const float *rb)
+{
+  return vec_vlxsspx (-32768, rb);
+}
+
+vf64_t
+test_vlxsspx_v0 (const signed long long ra, const float *rb)
+{
+  vf64_t xt;
+  __VEC_U_128 t;
+
+  float *p = (float *)((char *)rb + ra);
+  t.vf2[0] = t.vf2[1] = *p;
+  xt = t.vf2;
+  return xt;
+}
+
+void
+test_vstxsspx (vf64_t data, float *array, signed long offset)
+{
+  vec_vstxsspx (data, offset, array);
+}
+
+void
+test_vstxsspx_c0 (vf64_t data, float *array)
+{
+  vec_vstxsspx (data, 0, array);
+}
+
+void
+test_vstxsspx_c1 (vf64_t data, float *array)
+{
+  vec_vstxsspx (data, 8, array);
+}
+
+void
+test_vstxsspx_c2 (vf64_t data, float *array)
+{
+  vec_vstxsspx (data, 32760, array);
+}
+
+void
+test_vstxsspx_c3 (vf64_t data, float *array)
+{
+  vec_vstxsspx (data, 32768, array);
+}
+
+void
+test_vstxsspx_c4 (vf64_t data, float *array)
+{
+  vec_vstxsspx (data, -32768, array);
+}
+
+void
+test_vstxsspx_v1 (vf64_t data, float *array, signed long offset)
+{
+  __VEC_U_128 t;
+  float *p = (float *)((char *)array + offset);
+  t.vf2 = data;
+  *p = t.vf2[0];
+}
+
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+vf32_t
+test_vlxsspx_v1 (float *array, unsigned long offset)
+{
+  vf32_t res, rese;
+  vui8_t resp;
+
+  rese = vec_lvewx (offset, array);
+  resp = vec_lvsr (offset, array);
+  res = vec_perm (rese, rese, resp);
+  return res;
+}
+
+vf32_t
+test_vglssfso_v2 (float *array, unsigned long offset0, unsigned long offset1)
+{
+  vf32_t res, rese0, rese1;
+  vui8_t resp0, resp1;
+
+  rese0 = vec_lvewx (offset0, array);
+  rese1 = vec_lvewx (offset1, array);
+  resp0 = vec_lvsl (offset0, array);
+  resp1 = vec_lvsl (offset1, array);
+  rese0 = vec_perm (rese0, rese0, resp0);
+  rese1 = vec_perm (rese1, rese1, resp1);
+  res = vec_mergeh (rese0, rese1);
+  return res;
+}
+#endif
+
+vf32_t
+test_veclfgux_v1 (float *array, vui32_t idx)
+{
+  vf32_t res;
+  res[0] = array[idx[0]];
+  res[1] = array[idx[1]];
+  res[2] = array[idx[2]];
+  res[3] = array[idx[3]];
+  return res;
+}
+
+vf32_t
+test_veclfgux_v2 (float *array, vui32_t idx)
+{
+  __VEC_U_128 vidx, vflt;
+  vidx.vx4 = idx;
+  vflt.vf4[0] = array[vidx.uint.ix0];
+  vflt.vf4[1] = array[vidx.uint.ix1];
+  vflt.vf4[2] = array[vidx.uint.ix2];
+  vflt.vf4[3] = array[vidx.uint.ix3];
```

Comment:
I can't find members `vf4` and `uint`.  Am I missing another patch?
Maybe `test_veclfgux_v2()` should have not been included in this PR because it is not being called anywhere else.

Reply:
The "_dummy.c" sources are compile tests. I use those to try coding sequences and verify that pveclib "operations" with compile correctly. 99.99% are never called. Only a few are called from units / perf tests. 

This is more in the form of a failed experiment. The generated code was awful and toke different direction for the implementation. So I never commit the vflt/vidx struct changes for vec_common_ppc.h. I think deleting test_veclfgux_v2 is the best solution, asi it not used and a bad example.